sdk/src/types/block/semantic/state_transition.rs

```diff
@@ -387,10 +458,18 @@ impl StateTransitionVerifier for DelegationOutput {
     }
 
     fn destruction(
+        output_id: &OutputId,
         _current_state: &Self,
-        _context: &SemanticValidationContext<'_>,
+        context: &SemanticValidationContext<'_>,
     ) -> Result<(), StateTransitionError> {
-        // TODO handle mana rewards
+        // If a mana reward was provided but no reward context input exists
+        if context.mana_rewards.get(output_id).is_some() && !context.reward_context_inputs.contains_key(output_id) {
```

Comment:
What if there is no mana reward ?

Reply:
Then you're not claiming rewards I guess

Reply:
I feel like it would better detect mistakes if that was an error but alright

Reply:
Well, is it possible to destroy a delegation and not claim rewards?

Reply:
I don't think it's possible no. Just saying we could check the registered rewards actually match the destructions just in case

Reply:
Well if it's not possible then maybe we should not allow missing rewards full stop for destroyed delegations

Reply:
I feel like that's what I'm saying

Reply:
But let's merge for now, I need to go to bed